pkg/services/pluginsintegration/clientmiddleware/logger_middleware.go

```diff
@@ -37,25 +36,18 @@ type LoggerMiddleware struct {
 	features featuremgmt.FeatureToggles
 }
 
-func (m *LoggerMiddleware) logRequest(ctx context.Context, fn func(ctx context.Context) error) error {
-	status := statusOK
+func (m *LoggerMiddleware) logRequest(ctx context.Context, fn func(ctx context.Context) (requestStatus, error)) error {
 	start := time.Now()
 	timeBeforePluginRequest := log.TimeSinceStart(ctx, start)
 
-	err := fn(ctx)
-	if err != nil {
-		status = statusError
-		if errors.Is(err, context.Canceled) {
-			status = statusCancelled
-		}
-	}
+	status, err := fn(ctx)
 	logParams := []any{
 		"status", status,
 		"duration", time.Since(start),
 		"eventName", "grafana-data-egress",
 		"time_before_plugin_request", timeBeforePluginRequest,
 	}
-	if status == statusError {
+	if err != nil {
```

Comment:
Just noticed we're using info as log level below for any message - maybe we should use error level when status=error?

Reply:
sounds good to me as well!